tests/post/steps/test_ingress.py

```diff
@@ -0,0 +1,101 @@
+import json
+
+import requests
+import requests.exceptions
+
+import pytest
+from pytest_bdd import given, parsers, scenario, then, when
+
+
+@pytest.fixture(scope='function')
+def context():
+    return {}
+
+
+@scenario('../features/ingress.feature', 'Access HTTP services')
+def test_access_http_services(host):
+    pass
+
+
+@scenario('../features/ingress.feature', 'Access HTTPS services')
+def test_access_https_services(host):
+    pass
+
+
+@scenario(
+    '../features/ingress.feature', 'Access HTTP services on control-plane IP')
+def test_access_http_services_on_control_plane_ip(host):
+    pass
+
+
+@given("the node control-plane IP is not equal to its workload-plane IP")
+def node_control_plane_ip_is_not_equal_to_its_workload_plane_ip(host):
+    with host.sudo():
+        output = host.check_output(' '.join([
+            'salt-call --local',
+            'grains.get metalk8s',
+            '--out json',
+        ]))
+
+    data = json.loads(output)['local']
+
+    assert 'control_plane_ip' in data
+    assert 'workload_plane_ip' in data
+
+    if data['control_plane_ip'] == data['workload_plane_ip']:
+        pytest.skip('Node control-plane IP is equal to node workload-plane IP')
+
+
+@when(parsers.parse(
+    "we perform an {protocol} request on port {port} on a {plane} IP"))
+def perform_request(host, context, protocol, port, plane):
+    if protocol not in ['HTTP', 'HTTPS']:
+        raise NotImplementedError
+
+    grains = {
+        'workload-plane': 'metalk8s:workload_plane_ip',
+        'control-plane': 'metalk8s:control_plane_ip',
+    }
+
+    if plane not in grains:
+        raise NotImplementedError
+
+    with host.sudo():
+        ip_output = host.check_output(' '.join([
+            'salt-call --local',
+            'grains.get {grain}'.format(grain=grains[plane]),
+            '--out json',
+        ]))
+    ip = json.loads(ip_output)['local']
+
+    protocols = {
```

Comment:
If `protocols` is declared at the beginning of the function, you could rewrite the test l.52 as `if protocol not in protocols:`.

That would avoid the need to duplicate the list of supported protocols.

Edit: in fact the test l.52 seems useless as you are already doing what I suggest below.

Reply:
Or just use `protocol.lower()` instead of this mapping and keep the test l.52?